fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/JWTAuthMechanismFactory.java

```diff
@@ -120,22 +120,23 @@ public AuthenticationMechanism create(String mechanismName, FormParserFactory fo
                         jwksUri = readURLContent(jwksUriURL);
                     }
                 }
-                if (jwksUri == null) {
-                    throw new IllegalStateException("Neither a static key nor a JWKS URI was set.");
-                }
-                contextInfo.setJwksUri(jwksUri.trim());
+                if (jwksUri != null) {
+                    contextInfo.setJwksUri(jwksUri.trim());
 
-                String jwksRefreshInterval = properties.get("jwksRefreshInterval");
-                if (jwksRefreshInterval == null) {
-                    URL jwksRefreshIntervalURL = loader.getResource("/META-INF/MP-JWT-JWKS-REFRESH");
-                    if (jwksRefreshIntervalURL != null) {
-                        jwksRefreshInterval = readURLContent(jwksRefreshIntervalURL);
+                    String jwksRefreshInterval = properties.get("jwksRefreshInterval");
+                    if (jwksRefreshInterval == null) {
+                        URL jwksRefreshIntervalURL = loader.getResource("/META-INF/MP-JWT-JWKS-REFRESH");
+                        if (jwksRefreshIntervalURL != null) {
+                            jwksRefreshInterval = readURLContent(jwksRefreshIntervalURL);
+                        }
                     }
+                    if (jwksRefreshInterval == null) {
+                        throw new IllegalStateException("JWKS Refresh Interval should be set when JWKS URI is used.");
+                    }
+                    contextInfo.setJwksRefreshInterval(Integer.valueOf(jwksRefreshInterval.trim()));
+                } else {
+                    log.debug("Neither a static key nor a JWKS URI was set.");
```

Comment:
So this situation is OK? I mean -- the situation is probably OK when you integrate with Keycloak, but what if you don't and still omit both a static key and JWKS URI?

Reply:
This is why I added [this test](https://github.com/thorntail/thorntail/pull/1235/files#diff-baf5ffd0545003af243b14fd97938f31) to confirm that removing this check is safe even with the default code. Note it is part of the default microprofile jwt test suite. Also as I mentioned, this check is not actually done in the [portable configuration case](https://github.com/smallrye/smallrye-jwt/blob/master/implementation/src/main/java/io/smallrye/jwt/config/JWTAuthContextInfoProvider.java), as well as with [this provider](https://github.com/thorntail/thorntail/blob/master/fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/config/JWTAuthContextInfoProvider.java) due to [this code](https://github.com/thorntail/thorntail/blob/master/fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/JWTAuthMechanismFactory.java#L70). 
So it is safe to drop this check at this level, it is indirectly enforced anyway [in here](https://github.com/smallrye/smallrye-jwt/blob/master/implementation/src/main/java/io/smallrye/jwt/auth/principal/DefaultJWTCallerPrincipalFactory.java):
```
2019-02-25 15:38:55,931 INFO  [io.undertow.request.security] (default task-1) Failed to authenticate JWT bearer token
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.989 s - in org.wildfly.swarm.microprofile.jwtauth.NoVerificationKeyTest
```

Reply:
I'm not thinking correctness here, but user-friendliness. But I'm no expert on this code, so I'm gonna trust your judgement :-)